portal-bridge/src/bridge/state.rs

```diff
@@ -322,18 +330,57 @@ impl StateBridge {
     }
 }
 
-// Individual report for outcomes of offering a state content key
+/// Global report for outcomes of offering state content keys from long-running state bridge
+#[derive(Default)]
+struct GlobalOfferReport {
+    success: usize,
+    failed: usize,
+    declined: usize,
+}
+
+impl GlobalOfferReport {
+    fn update(&mut self, trace: &OfferTrace) {
+        match trace {
+            OfferTrace::Success(_) => self.success += 1,
+            OfferTrace::Failed => self.failed += 1,
+            OfferTrace::Declined => self.declined += 1,
+        }
+    }
+
+    fn report(&self) {
+        let total = self.success + self.failed + self.declined;
+        info!(
+            "State offer report: Total Offers: {}. Successful: {}% ({}). Declined: {}% ({}). Failed: {}% ({}).",
+            total,
+            format!("{:.2}", calculate_percentage(self.success, total)),
+            self.success,
+            format!("{:.2}", calculate_percentage(self.declined, total)),
+            self.declined,
+            format!("{:.2}", calculate_percentage(self.failed, total)),
+            self.failed,
+        );
+    }
+}
+
+fn calculate_percentage(part: usize, total: usize) -> f64 {
```

Comment:
Seems like this belongs in a utility module. I expected to find a humanize or numbers module in trin-utils, but apparently we don't have one yet.